exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/logs/OtlpGrpcLogRecordExporterBuilder.java

```diff
@@ -109,15 +114,22 @@ public OtlpGrpcLogRecordExporterBuilder setEndpoint(String endpoint) {
   }
 
   /**
-   * Sets the method used to compress payloads. If unset, compression is disabled. Currently
-   * supported compression methods include "gzip" and "none".
+   * Sets the method used to compress payloads. If unset, compression is disabled. Compression
+   * method "gzip" and "none" are supported out of the box. Support for additional compression
+   * methods is available by implementing {@link Compressor} and {@link CompressorProvider}.
    */
   public OtlpGrpcLogRecordExporterBuilder setCompression(String compressionMethod) {
     requireNonNull(compressionMethod, "compressionMethod");
+    if (compressionMethod.equals("none")) {
+      delegate.setCompression(null);
+      return this;
+    }
+    Set<String> supportedCompressionMethods = CompressorUtil.supportedCompressors();
     checkArgument(
-        compressionMethod.equals("gzip") || compressionMethod.equals("none"),
-        "Unsupported compression method. Supported compression methods include: gzip, none.");
-    delegate.setCompression(compressionMethod);
+        supportedCompressionMethods.contains(compressionMethod),
+        "Unsupported compressionMethod. Compression method must be \"none\" or one of: "
```

Comment:
I know there was existing art here, but I could also see empty string being a semantic equivalence for "none". Can happen in a future PR tho if folks like that.

Reply:
The `none` keyword comes directly from the [spec](https://github.com/open-telemetry/opentelemetry-specification/blame/a6ca2fd484c9e76fe1d8e1c79c99f08f4745b5ee/specification/protocol/exporter.md#L81).

I don't like the idea of having two ways to accomplish the same thing, but if we were to add an alternative, `null` makes more sense to me than an empty string.

Reply:
Well, this is certainly a larger conversation then about spec and style. One that I don't need to be dragged out here.  😁 Heh.